File: ModiFE/lib/models/res/shop_reses.dart
import 'package:modi/models/res/base_res.dart';
import 'package:modi/models/single_note.dart';

class GetPricesRes extends BaseRes {
  int randomNote;
  int randomMonthNote;
  int note;

  GetPricesRes.fromJson(Map<String, dynamic> json) :
    randomNote = json["randomNote"] ?? 0,
    randomMonthNote = json["randomMonthNote"] ?? 0,
    note = json["note"] ?? 0,
    super.fromJson(json);
}

class BuyNoteRes extends BaseRes {
  SingleNote? unlockedNote;
  int tokens;

  BuyNoteRes.fromJson(Map<String, dynamic> json) :
    unlockedNote = json["unlockedNote"] != null ? SingleNote.fromJson(json["unlockedNote"]) : null,
    tokens = json["tokens"] ?? 0,
    super.fromJson(json);
}

File: ModiBE/src/models/res/shop-reses.ts
import { BaseRes } from "./base-res";
import { SingleNote } from "./note-reses";

export class BuyNoteRes extends BaseRes {
  public unlockedNote: SingleNote;
  public tokens: number;

  constructor(unlockedNote: SingleNote, tokens: number) {
    super("", false);
    this.unlockedNote = unlockedNote;
    this.tokens = tokens;
  }
}

export class GetPricesRes extends BaseRes {
  public randomNote: number;
  public randomMonthNote: number;
  public note: number;

  constructor(randomNote: number, randomMonthNote: number, note: number) {
    super("", false);
    this.randomNote = randomNote;
    this.randomMonthNote = randomMonthNote;
    this.note = note;
  }
}
